pyt/cfg/stmt_visitor.py

```diff
@@ -555,23 +555,44 @@ def visit_For(self, node):
             path=self.filenames[-1]
         ))
 
-        if isinstance(node.iter, ast.Call) and get_call_names_as_string(node.iter.func) in self.function_names:
-            last_node = self.visit(node.iter)
-            last_node.connect(for_node)
+        self.process_loop_funcs(node.iter, for_node)
 
         return self.loop_node_skeleton(for_node, node)
 
+    def process_loop_funcs(self, comp_n, loop_node):
+        """
+        If the loop test node contains function calls, it connects the loop node to the nodes of
+        those function calls.
+
+        :param comp_n: The test node of a loop that may contain functions.
+        :param loop_node: The loop node itself to connect to the new function nodes if any
+        :return: None
+        """
+        if isinstance(comp_n, ast.Call) and get_call_names_as_string(comp_n.func) in self.function_names:
```

Comment:
I know this was already here, but I don't understand why `get_call_names_as_string(comp_n.func) in self.function_names` is there. It means that `range` is ignored for instance. I'd be tempted to remove it, but maybe in a later PR.

Reply:
I didn't pay much mind to this detail but if removed, 4 of the test cases or `for` fail due to functions in the iter node being ignored (e.g. `range`). 

I think the rationale here is to only link the while / for node to the function node if we've visited the function before, but I'm not sure why that would matter since we're immediately visiting the call node. 

That said, we don't have a `visit_Call` method, so the default visitor will be invoked, which will in turn call `visit_Name` (which we don't have, so default again), then I think a call to `visit_Load` which will be default again, etc. I wonder if we'd need a `visit_Call` to handle this case before we can safely remove the `in self.function_names`.